test/e2e/create_test.go

```diff
@@ -455,29 +455,23 @@ func (tc *testContext) listFullyConfiguredWindowsNodes(isBYOH bool) ([]v1.Node,
 	return windowsNodes, nil
 }
 
-// scaleMachineApproverDeployment scales the Machine Approver deployment pods to the expectedPodCount
-func (tc *testContext) scaleMachineApproverDeployment(expectedPodCount *int32) error {
-	deployment, err := tc.client.K8s.AppsV1().Deployments("openshift-cluster-machine-approver").Get(context.TODO(),
-		"machine-approver", metav1.GetOptions{})
+// scaleDeployment scales the deployment associated with the given namespace and name to the expectedPodCount
+func (tc *testContext) scaleDeployment(namespace, name string, expectedPodCount *int32) error {
+	deployment, err := tc.client.K8s.AppsV1().Deployments(namespace).Get(context.TODO(), name, metav1.GetOptions{})
 	if err != nil {
-		return errors.Wrap(err, "error listing Cluster Machine Approver deployment")
+		return errors.Wrapf(err, "error listing deployment %s/%s", namespace, name)
 	}
 
 	deployment.Spec.Replicas = expectedPodCount
-	_, err = tc.client.K8s.AppsV1().Deployments("openshift-cluster-machine-approver").Update(context.TODO(),
-		deployment, metav1.UpdateOptions{})
+	_, err = tc.client.K8s.AppsV1().Deployments(namespace).Update(context.TODO(), deployment, metav1.UpdateOptions{})
 	if err != nil {
-		return errors.Wrap(err, "error updating Cluster Machine Approver deployment")
+		return errors.Wrapf(err, "error updating deployment %s/%s", namespace, name)
 	}
-	retryInterval := retry.Interval
-	retryTimeout := retry.Timeout
-	if err = wait.Poll(retryInterval, retryTimeout, func() (bool, error) {
-		if deployment.Spec.Replicas == expectedPodCount {
-			return true, nil
-		}
-		return false, nil
+
+	if err = wait.Poll(retry.Interval, retry.Timeout, func() (bool, error) {
+		return deployment.Spec.Replicas == expectedPodCount, nil
 	}); err != nil {
```

Comment:
How is this poll working? Unless you redo the get of the deployment, the value is not going to change.

Reply:
Shouldn't this also be looking at the status and not the spec...?
The spec is always going to be changed by the update, we need to wait for the actual count of replicas to reach zero.
[api reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#deployment-v1-apps)

Is this why the bug is occurring? :thinking:

Reply:
I'll be honest, it looked off but I didn't want to touch the logic since in the past and through all my testing it seemed to work. The CMA deployment is being scaled down properly, the issue is that it comes back up despite our CVO override. So I doubt its causing the bug, but I will edit the logic with your suggestions.